src/View/mainpage/DeleteROIWindow.py

```diff
@@ -304,13 +307,50 @@ def confirm_button_onClicked(self):
                                                       'Would you like to continue?',
                                                       QMessageBox.Yes | QMessageBox.No)
         if confirmation_dialog == QMessageBox.Yes:
-            for item in self.regions_of_interest_to_delete:
-                new_dataset = ROI.delete_roi(self.dataset_rtss, item)
-
-            self.deleting_rois_structure_tuple.emit((new_dataset, {"delete": self.regions_of_interest_to_delete}))
-            self.close()
-
-
-
+            progress_window = DeleteROIProgressWindow(self, QtCore.Qt.WindowTitleHint)
+            progress_window.signal_roi_deleted.connect(self.on_rois_deleted)
+            progress_window.start_deleting(self.dataset_rtss, self.regions_of_interest_to_delete)
```

Comment:
is the dialog modal?
if not, what happens if someone starts doing other things while waiting for the ROIs to be fully deleted?  Are the ROIs contained in something thread safe?

Reply:
It is modal, yes. The first parameter of the constructor (in this case `self`) is the parent window of the window.